fractal_tasks_core/tasks/illumination_correction.py

```diff
@@ -289,7 +290,7 @@ def illumination_correction(
     # pyramid of coarser levels
     build_pyramid(
         zarrurl=fov_path,
-        overwrite=overwrite,
+        overwrite=overwrite_input,
```

Comment:
This is not related to input overwriting. Can be set the same as the overwrite just above (e.g. hard-coded to False) as long as we only allow overwrite_input = True. If we implement overwrite_input = False to also work, then this is where the typical overwrite functionality (for output overwriting) would come into effect.

Reply:
> This is not related to input overwriting. Can be set the same as the overwrite just above (e.g. hard-coded to False) as long as we only allow overwrite_input = True. If we implement overwrite_input = False to also work, then this is where the typical overwrite functionality (for output overwriting) would come into effect.

I think you got it the other way round.

For the moment we only allow `overwrite_input=True`, and then we also have to set `overwrite=True` in the pyramid function. If we set `overwrite=False`, then this task will always fail:
1. We have an existing zarr, with pyramids
2. We call illumination correction with `overwrite_input=True` (the only supported one)
3. Level 0 is overwritten (as it should)
4. When building pyramids, we try to overwrite existing level 1 and fail.

To recap
* In the current version, we only support `overwrite_input=True`, and we need `overwrite=True`.
* In a future version where we support `overwrite_input=False`, we will have to decide what to do when the new component already exists. In principle the task may have two parameters `overwrite_input` and `overwrite`, where the latter determines whether _outputs_ should be overwritten. This discussion is currently out of scope.

TLDR: I am now hard-coding this pyramid-building parameter to `overwrite=True`, @jluethi let me know if this makes sense to you.

Reply:
> In a future version where we support overwrite_input=False, we will have to decide what to do when the new component already exists. In principle the task may have two parameters overwrite_input and overwrite, where the latter determines whether outputs should be overwritten. This discussion is currently out of scope.

Agreed.

overwrite=True hard coded makes sense given this logic! :)